ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/TextEmbeddingModel.java

```diff
@@ -36,6 +46,52 @@ public ModelTensorOutput predict(String modelId, MLInput mlInput) throws Transla
         return new ModelTensorOutput(tensorOutputs);
     }
 
+    private boolean isAsymmetricModel(MLAlgoParams mlParams) {
+        if (mlParams instanceof AsymmetricTextEmbeddingParameters) {
+            // Check for the necessary prefixes in modelConfig
+            if (modelConfig == null
```

Comment:
I think we need to refine this condition.

1. What if model config is not null, but in that model config passage prefix and query prefix isn't provided?
2. Shouldn't we expect both prefixes in the model config, so that later there's no confusion?

Reply:
Your point 1. is precisely what is being checked for in this first `if` statement. 

If the model config is not null, but both passage and query prefixes are not provided, we throw an `IllegalArgumentException` saying the model needs to have at least one of them, if you want to use `AsymmetricTextEmbeddingParameters`.

Regarding 2: I don't know any asymmetric model that only requires **one** of both prefixes, but this doesn't mean there isn't one. It would actually be a pretty reasonable design decision to have `passage` be the default. Because of this I thought that it would make sense to be less strict, since we don't really know how these models will develop in the future. 

If you think this is very important, I can change the condition and require both prefixes to be set. If a model only requires one of them, the user would be required to set the empty string as prefix for the other.

Reply:
In this condition, we are checking both prefixes (&&), but in the message we are telling either one of these two needs to be set. 


We usually use the same model for query & passage, so from this context both should be present. But now in the second thought, there's no cons to keep this flexible, right?

Reply:
We're checking if both (`&&`) are `null`, and if so, throwing the exception telling the at least one of them needs to be **not** `null`.

Reply:
Yes, I think it's better to keep it flexible.

Reply:
for an example of a model that uses a query prefix and no passage prefix: https://huggingface.co/BAAI/bge-large-en#frequently-asked-questions